app.go

```diff
@@ -261,22 +262,67 @@ func (t stopTimeoutOption) String() string {
 	return fmt.Sprintf("fx.StopTimeout(%v)", time.Duration(t))
 }
 
+// withLogger will stay private until we export fxlog.Logger once
+// we finalize the API.
+func withLogger(l fxlog.Logger) Option {
+	return withLoggerOption{l}
+}
+
+type withLoggerOption struct{ l fxlog.Logger }
+
+func (l withLoggerOption) apply(app *App) {
+	app.log = l.l
+}
+
+func (l withLoggerOption) String() string {
+	return fmt.Sprintf("fx.withLogger(%v)", l.l)
+}
+
 // Printer is the interface required by Fx's logging backend. It's implemented
 // by most loggers, including the one bundled with the standard library.
+//
+// Note, this will be deprecate with next release and you will need to implement
+// fxlog.Logger interface instead.
 type Printer interface {
 	Printf(string, ...interface{})
 }
 
 // Logger redirects the application's log output to the provided printer.
+//
+// Note, this will be deprecated with the next release and you will need to use
+// withLogger instead.
 func Logger(p Printer) Option {
 	return loggerOption{p}
 }
 
 type loggerOption struct{ p Printer }
 
+type printerWrapper struct {
+	p Printer
+}
+
+// newPrinter returns an implementation of zapcore.WriteSyncer
+// used to support Logger option which implements Printer
+// interface.
+func newPrinter(p Printer) zapcore.WriteSyncer {
+	return &printerWrapper{
+		p: p,
```

Comment:
I would move this and its implementation either to its own file (`printer.go`?) or into fxlog (`fxlog.WriteSyncerFrom(Printer)`?).